pkg/storaged/test-util.js

```diff
@@ -56,4 +56,36 @@ QUnit.test("compare_versions", function() {
     }
 });
 
+QUnit.test("mdraid_name_nohostnamed", function() {
+    // the proxy should not be valid yet; if it races us, skip the test
+    var res = utils.mdraid_name({ "Name": "somehost:mydev" });
+    if (utils.hostnamed.StaticHostname === undefined)
+        assert.strictEqual(res, "mydev", "remote host name is skipped before dbus proxy initializes");
+});
+
+QUnit.asyncTest("mdraid_name_remote", function() {
+    assert.expect(1);
+
+    utils.hostnamed.wait(function() {
```

Comment:
I think it is confusing that we wait for hostnamed here and then not use it.  The reason to wait for it is to make sure that the whole test run doesn't end before the hostnamed proxy is ready.  So what about putting the wait at the bottom of this file and wrap the first QUnit.start() call?

Also, the reason why we have to wait for the proxy to be ready is quite obscure, no?  A comment should explain that, I'd say.